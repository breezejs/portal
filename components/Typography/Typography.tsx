import React, {createElement, Fragment, memo, ReactNode} from 'react';
import styled from 'styled-components';
import {GlobalStylesTheme} from '../../core/GlobalStyles';

function buildComponent (props: ITypographyProps) {
  const coreHeadingStyles = `
    color: ${GlobalStylesTheme.headingsColour};
    font-family: ${GlobalStylesTheme.headingsFontFamily};
    font-weight: ${GlobalStylesTheme.headingsFontWeight};
    line-height: ${GlobalStylesTheme.headingsLineHeight};
    margin-bottom: ${GlobalStylesTheme.headingsMarginBottom};
  `;
  const variants = {
    h1: `
      font-size: ${GlobalStylesTheme.h1FontSize};
      ${coreHeadingStyles}
    `,
    h2: `
      font-size: ${GlobalStylesTheme.h2FontSize};
      ${coreHeadingStyles}
    `,
    h3: `
      font-size: ${GlobalStylesTheme.h3FontSize};
      ${coreHeadingStyles}
    `,
    h4: `
      font-size: ${GlobalStylesTheme.h4FontSize};
      ${coreHeadingStyles}
    `,
    h5: `
      font-size: ${GlobalStylesTheme.h5FontSize};
      ${coreHeadingStyles}
    `,
    h6: `
      font-size: ${GlobalStylesTheme.h6FontSize};
      ${coreHeadingStyles}
    `,
    lead: `
      font-size: ${GlobalStylesTheme.leadFontSize};
      font-weight: ${GlobalStylesTheme.leadFontWeight};
    `,
    p: `
      font-size: ${GlobalStylesTheme.fontSizeBase};
    `
  };
  const Component = styled(props.component)`
    ${({variant}: ITypographyProps) => variants[variant]}
  `;

  return createElement(Component, props);
}

type TypographyComponentTypes =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'h5'
  | 'h6'
  | 'p';

type TypographyVariantTypes =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'h5'
  | 'h6'
  | 'lead'
  | 'p';

interface ITypographyProps {
  children: ReactNode;
  component: TypographyComponentTypes;
  variant: TypographyVariantTypes;
}

function Typography (props: ITypographyProps) {
  return (
    <Fragment>{buildComponent(props)}</Fragment>
  );
}

export default memo(Typography);
