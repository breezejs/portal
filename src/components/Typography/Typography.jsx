import PropTypes from 'prop-types';
import React, {createElement, Fragment, memo} from 'react';
import styled from 'styled-components';
import {GlobalStylesTheme} from '../GlobalStyles';

function buildComponent (properties) {
  const coreHeadingStyles = `
    color: ${GlobalStylesTheme.headingsColour};
    font-family: ${GlobalStylesTheme.headingsFontFamily};
    font-weight: ${GlobalStylesTheme.headingsFontWeight};
    line-height: ${GlobalStylesTheme.headingsLineHeight};
    margin-bottom: ${GlobalStylesTheme.headingsMarginBottom};
  `;
  const variants = {
    h1: `
      font-size: ${GlobalStylesTheme.h1FontSize};
      ${coreHeadingStyles}
    `,
    h2: `
      font-size: ${GlobalStylesTheme.h2FontSize};
      ${coreHeadingStyles}
    `,
    h3: `
      font-size: ${GlobalStylesTheme.h3FontSize};
      ${coreHeadingStyles}
    `,
    h4: `
      font-size: ${GlobalStylesTheme.h4FontSize};
      ${coreHeadingStyles}
    `,
    h5: `
      font-size: ${GlobalStylesTheme.h5FontSize};
      ${coreHeadingStyles}
    `,
    h6: `
      font-size: ${GlobalStylesTheme.h6FontSize};
      ${coreHeadingStyles}
    `,
    lead: `
      font-size: ${GlobalStylesTheme.leadFontSize};
      font-weight: ${GlobalStylesTheme.leadFontWeight};
    `,
    p: `
      font-size: ${GlobalStylesTheme.fontSizeBase};
    `
  };
  const {component} = properties;
  const Component = styled(component)`
    ${({variant}) => variants[variant]}
  `;

  Component.propTypes = {};

  return createElement(Component, properties);
}

function Typography (props) {
  return (
    <Fragment>{buildComponent(props)}</Fragment>
  );
}

Typography.propTypes = {
  children: PropTypes.node.isRequired,
  component: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p']).isRequired,
  variant: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'lead', 'p']).isRequired
};

export default memo(Typography);
