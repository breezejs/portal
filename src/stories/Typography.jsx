import {optionsKnob, withKnobs} from '@storybook/addon-knobs';
import {storiesOf} from '@storybook/react';
import React from 'react';
import GlobalStyles from '../components/GlobalStyles';
import Typography from '../components/Typography';

const stories = storiesOf('Typography', module);

stories
  .addDecorator(withKnobs)
  .addDecorator((storyFn) => (
    <div>
      <GlobalStyles />
      {storyFn()}
    </div>
  ))
  .add('with H1 element and variant', () => (
    <Typography component="h1" variant="h1">Typography</Typography>
  ))
  .add('with H2 element and variant', () => (
    <Typography component="h2" variant="h2">Typography</Typography>
  ))
  .add('with H3 element and variant', () => (
    <Typography component="h3" variant="h3">Typography</Typography>
  ))
  .add('with H4 element and variant', () => (
    <Typography component="h4" variant="h4">Typography</Typography>
  ))
  .add('with H5 element and variant', () => (
    <Typography component="h5" variant="h5">Typography</Typography>
  ))
  .add('with H6 element and variant', () => (
    <Typography component="h6" variant="h6">Typography</Typography>
  ))
  .add('with P element and variant', () => (
    <Typography component="p" variant="p">Typography</Typography>
  ))
  .add('with P element and lead variant', () => (
    <Typography component="p" variant="lead">Typography</Typography>
  ))
  .add('with full customisation', () => {
    const componentOptions = {
      h1: 'h1',
      h2: 'h2',
      h3: 'h3',
      h4: 'h4',
      h5: 'h5',
      h6: 'h6',
      p: 'p'
    };
    const variantOptions = {
      h1: 'h1',
      h2: 'h2',
      h3: 'h3',
      h4: 'h4',
      h5: 'h5',
      h6: 'h6',
      lead: 'lead',
      p: 'p'
    };
    const defaultOption = 'h1';
    const optionsObj = {
      display: 'select'
    };

    return (
      <Typography
        component={optionsKnob('Component', componentOptions, defaultOption, optionsObj)}
        variant={optionsKnob('Variant', variantOptions, defaultOption, optionsObj)}
      >Typography</Typography>
    );
  });
