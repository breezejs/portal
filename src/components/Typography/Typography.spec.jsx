import {mount} from 'enzyme';
import React from 'react';
import Typography from './Typography';

describe('Typography', () => {
  describe('h1 element', () => {
    it('should render as h1 variant style', () => {
      const result = mount(
        <Typography component='h1' variant='h1'>Test</Typography>
      );

      expect(result.find('h1')).toMatchSnapshot();
    });

    it('should render as h2 variant style', () => {
      const result = mount(
        <Typography component='h1' variant='h2'>Test</Typography>
      );

      expect(result.find('h1')).toMatchSnapshot();
    });

    it('should render as h3 variant style', () => {
      const result = mount(
        <Typography component='h1' variant='h3'>Test</Typography>
      );

      expect(result.find('h1')).toMatchSnapshot();
    });

    it('should render as h4 variant style', () => {
      const result = mount(
        <Typography component='h1' variant='h4'>Test</Typography>
      );

      expect(result.find('h1')).toMatchSnapshot();
    });

    it('should render as h5 variant style', () => {
      const result = mount(
        <Typography component='h1' variant='h5'>Test</Typography>
      );

      expect(result.find('h1')).toMatchSnapshot();
    });

    it('should render as h6 variant style', () => {
      const result = mount(
        <Typography component='h1' variant='h6'>Test</Typography>
      );

      expect(result.find('h1')).toMatchSnapshot();
    });

    it('should render as lead variant style', () => {
      const result = mount(
        <Typography component='h1' variant='lead'>Test</Typography>
      );

      expect(result.find('h1')).toMatchSnapshot();
    });

    it('should render as p variant style', () => {
      const result = mount(
        <Typography component='h1' variant='p'>Test</Typography>
      );

      expect(result.find('h1')).toMatchSnapshot();
    });
  });

  describe('h2 element', () => {
    it('should render as h1 variant style', () => {
      const result = mount(
        <Typography component='h2' variant='h1'>Test</Typography>
      );

      expect(result.find('h2')).toMatchSnapshot();
    });

    it('should render as h2 variant style', () => {
      const result = mount(
        <Typography component='h2' variant='h2'>Test</Typography>
      );

      expect(result.find('h2')).toMatchSnapshot();
    });

    it('should render as h3 variant style', () => {
      const result = mount(
        <Typography component='h2' variant='h3'>Test</Typography>
      );

      expect(result.find('h2')).toMatchSnapshot();
    });

    it('should render as h4 variant style', () => {
      const result = mount(
        <Typography component='h2' variant='h4'>Test</Typography>
      );

      expect(result.find('h2')).toMatchSnapshot();
    });

    it('should render as h5 variant style', () => {
      const result = mount(
        <Typography component='h2' variant='h5'>Test</Typography>
      );

      expect(result.find('h2')).toMatchSnapshot();
    });

    it('should render as h6 variant style', () => {
      const result = mount(
        <Typography component='h2' variant='h6'>Test</Typography>
      );

      expect(result.find('h2')).toMatchSnapshot();
    });

    it('should render as lead variant style', () => {
      const result = mount(
        <Typography component='h2' variant='lead'>Test</Typography>
      );

      expect(result.find('h2')).toMatchSnapshot();
    });

    it('should render as p variant style', () => {
      const result = mount(
        <Typography component='h2' variant='p'>Test</Typography>
      );

      expect(result.find('h2')).toMatchSnapshot();
    });
  });

  describe('h3 element', () => {
    it('should render as h1 variant style', () => {
      const result = mount(
        <Typography component='h3' variant='h1'>Test</Typography>
      );

      expect(result.find('h3')).toMatchSnapshot();
    });

    it('should render as h2 variant style', () => {
      const result = mount(
        <Typography component='h3' variant='h2'>Test</Typography>
      );

      expect(result.find('h3')).toMatchSnapshot();
    });

    it('should render as h3 variant style', () => {
      const result = mount(
        <Typography component='h3' variant='h3'>Test</Typography>
      );

      expect(result.find('h3')).toMatchSnapshot();
    });

    it('should render as h4 variant style', () => {
      const result = mount(
        <Typography component='h3' variant='h4'>Test</Typography>
      );

      expect(result.find('h3')).toMatchSnapshot();
    });

    it('should render as h5 variant style', () => {
      const result = mount(
        <Typography component='h3' variant='h5'>Test</Typography>
      );

      expect(result.find('h3')).toMatchSnapshot();
    });

    it('should render as h6 variant style', () => {
      const result = mount(
        <Typography component='h3' variant='h6'>Test</Typography>
      );

      expect(result.find('h3')).toMatchSnapshot();
    });

    it('should render as lead variant style', () => {
      const result = mount(
        <Typography component='h3' variant='lead'>Test</Typography>
      );

      expect(result.find('h3')).toMatchSnapshot();
    });

    it('should render as p variant style', () => {
      const result = mount(
        <Typography component='h3' variant='p'>Test</Typography>
      );

      expect(result.find('h3')).toMatchSnapshot();
    });
  });

  describe('h4 element', () => {
    it('should render as h1 variant style', () => {
      const result = mount(
        <Typography component='h4' variant='h1'>Test</Typography>
      );

      expect(result.find('h4')).toMatchSnapshot();
    });

    it('should render as h2 variant style', () => {
      const result = mount(
        <Typography component='h4' variant='h2'>Test</Typography>
      );

      expect(result.find('h4')).toMatchSnapshot();
    });

    it('should render as h3 variant style', () => {
      const result = mount(
        <Typography component='h4' variant='h3'>Test</Typography>
      );

      expect(result.find('h4')).toMatchSnapshot();
    });

    it('should render as h4 variant style', () => {
      const result = mount(
        <Typography component='h4' variant='h4'>Test</Typography>
      );

      expect(result.find('h4')).toMatchSnapshot();
    });

    it('should render as h5 variant style', () => {
      const result = mount(
        <Typography component='h4' variant='h5'>Test</Typography>
      );

      expect(result.find('h4')).toMatchSnapshot();
    });

    it('should render as h6 variant style', () => {
      const result = mount(
        <Typography component='h4' variant='h6'>Test</Typography>
      );

      expect(result.find('h4')).toMatchSnapshot();
    });

    it('should render as lead variant style', () => {
      const result = mount(
        <Typography component='h4' variant='lead'>Test</Typography>
      );

      expect(result.find('h4')).toMatchSnapshot();
    });

    it('should render as p variant style', () => {
      const result = mount(
        <Typography component='h4' variant='p'>Test</Typography>
      );

      expect(result.find('h4')).toMatchSnapshot();
    });
  });

  describe('h5 element', () => {
    it('should render as h1 variant style', () => {
      const result = mount(
        <Typography component='h5' variant='h1'>Test</Typography>
      );

      expect(result.find('h5')).toMatchSnapshot();
    });

    it('should render as h2 variant style', () => {
      const result = mount(
        <Typography component='h5' variant='h2'>Test</Typography>
      );

      expect(result.find('h5')).toMatchSnapshot();
    });

    it('should render as h3 variant style', () => {
      const result = mount(
        <Typography component='h5' variant='h3'>Test</Typography>
      );

      expect(result.find('h5')).toMatchSnapshot();
    });

    it('should render as h4 variant style', () => {
      const result = mount(
        <Typography component='h5' variant='h4'>Test</Typography>
      );

      expect(result.find('h5')).toMatchSnapshot();
    });

    it('should render as h5 variant style', () => {
      const result = mount(
        <Typography component='h5' variant='h5'>Test</Typography>
      );

      expect(result.find('h5')).toMatchSnapshot();
    });

    it('should render as h6 variant style', () => {
      const result = mount(
        <Typography component='h5' variant='h6'>Test</Typography>
      );

      expect(result.find('h5')).toMatchSnapshot();
    });

    it('should render as lead variant style', () => {
      const result = mount(
        <Typography component='h5' variant='lead'>Test</Typography>
      );

      expect(result.find('h5')).toMatchSnapshot();
    });

    it('should render as p variant style', () => {
      const result = mount(
        <Typography component='h5' variant='p'>Test</Typography>
      );

      expect(result.find('h5')).toMatchSnapshot();
    });
  });

  describe('h6 element', () => {
    it('should render as h1 variant style', () => {
      const result = mount(
        <Typography component='h6' variant='h1'>Test</Typography>
      );

      expect(result.find('h6')).toMatchSnapshot();
    });

    it('should render as h2 variant style', () => {
      const result = mount(
        <Typography component='h6' variant='h2'>Test</Typography>
      );

      expect(result.find('h6')).toMatchSnapshot();
    });

    it('should render as h3 variant style', () => {
      const result = mount(
        <Typography component='h6' variant='h3'>Test</Typography>
      );

      expect(result.find('h6')).toMatchSnapshot();
    });

    it('should render as h4 variant style', () => {
      const result = mount(
        <Typography component='h6' variant='h4'>Test</Typography>
      );

      expect(result.find('h6')).toMatchSnapshot();
    });

    it('should render as h5 variant style', () => {
      const result = mount(
        <Typography component='h6' variant='h5'>Test</Typography>
      );

      expect(result.find('h6')).toMatchSnapshot();
    });

    it('should render as h6 variant style', () => {
      const result = mount(
        <Typography component='h6' variant='h6'>Test</Typography>
      );

      expect(result.find('h6')).toMatchSnapshot();
    });

    it('should render as lead variant style', () => {
      const result = mount(
        <Typography component='h6' variant='lead'>Test</Typography>
      );

      expect(result.find('h6')).toMatchSnapshot();
    });

    it('should render as p variant style', () => {
      const result = mount(
        <Typography component='h6' variant='p'>Test</Typography>
      );

      expect(result.find('h6')).toMatchSnapshot();
    });
  });

  describe('p element', () => {
    it('should render as h1 variant style', () => {
      const result = mount(
        <Typography component='p' variant='h1'>Test</Typography>
      );

      expect(result.find('p')).toMatchSnapshot();
    });

    it('should render as h2 variant style', () => {
      const result = mount(
        <Typography component='p' variant='h2'>Test</Typography>
      );

      expect(result.find('p')).toMatchSnapshot();
    });

    it('should render as h3 variant style', () => {
      const result = mount(
        <Typography component='p' variant='h3'>Test</Typography>
      );

      expect(result.find('p')).toMatchSnapshot();
    });

    it('should render as h4 variant style', () => {
      const result = mount(
        <Typography component='p' variant='h4'>Test</Typography>
      );

      expect(result.find('p')).toMatchSnapshot();
    });

    it('should render as h5 variant style', () => {
      const result = mount(
        <Typography component='p' variant='h5'>Test</Typography>
      );

      expect(result.find('p')).toMatchSnapshot();
    });

    it('should render as h6 variant style', () => {
      const result = mount(
        <Typography component='p' variant='h6'>Test</Typography>
      );

      expect(result.find('p')).toMatchSnapshot();
    });

    it('should render as lead variant style', () => {
      const result = mount(
        <Typography component='p' variant='lead'>Test</Typography>
      );

      expect(result.find('p')).toMatchSnapshot();
    });

    it('should render as p variant style', () => {
      const result = mount(
        <Typography component='p' variant='p'>Test</Typography>
      );

      expect(result.find('p')).toMatchSnapshot();
    });
  });
});
