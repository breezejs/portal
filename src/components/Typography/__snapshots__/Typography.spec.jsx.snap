// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography h1 element should render as h1 variant style 1`] = `
.c0 {
  font-size: 2.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h1
  className="c0"
>
  Test
</h1>
`;

exports[`Typography h1 element should render as h2 variant style 1`] = `
.c0 {
  font-size: 2rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h1
  className="c0"
>
  Test
</h1>
`;

exports[`Typography h1 element should render as h3 variant style 1`] = `
.c0 {
  font-size: 1.75rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h1
  className="c0"
>
  Test
</h1>
`;

exports[`Typography h1 element should render as h4 variant style 1`] = `
.c0 {
  font-size: 1.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h1
  className="c0"
>
  Test
</h1>
`;

exports[`Typography h1 element should render as h5 variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h1
  className="c0"
>
  Test
</h1>
`;

exports[`Typography h1 element should render as h6 variant style 1`] = `
.c0 {
  font-size: 1rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h1
  className="c0"
>
  Test
</h1>
`;

exports[`Typography h1 element should render as lead variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  font-weight: 300;
}

<h1
  className="c0"
>
  Test
</h1>
`;

exports[`Typography h1 element should render as p variant style 1`] = `
.c0 {
  font-size: 1rem;
}

<h1
  className="c0"
>
  Test
</h1>
`;

exports[`Typography h2 element should render as h1 variant style 1`] = `
.c0 {
  font-size: 2.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h2
  className="c0"
>
  Test
</h2>
`;

exports[`Typography h2 element should render as h2 variant style 1`] = `
.c0 {
  font-size: 2rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h2
  className="c0"
>
  Test
</h2>
`;

exports[`Typography h2 element should render as h3 variant style 1`] = `
.c0 {
  font-size: 1.75rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h2
  className="c0"
>
  Test
</h2>
`;

exports[`Typography h2 element should render as h4 variant style 1`] = `
.c0 {
  font-size: 1.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h2
  className="c0"
>
  Test
</h2>
`;

exports[`Typography h2 element should render as h5 variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h2
  className="c0"
>
  Test
</h2>
`;

exports[`Typography h2 element should render as h6 variant style 1`] = `
.c0 {
  font-size: 1rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h2
  className="c0"
>
  Test
</h2>
`;

exports[`Typography h2 element should render as lead variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  font-weight: 300;
}

<h2
  className="c0"
>
  Test
</h2>
`;

exports[`Typography h2 element should render as p variant style 1`] = `
.c0 {
  font-size: 1rem;
}

<h2
  className="c0"
>
  Test
</h2>
`;

exports[`Typography h3 element should render as h1 variant style 1`] = `
.c0 {
  font-size: 2.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h3
  className="c0"
>
  Test
</h3>
`;

exports[`Typography h3 element should render as h2 variant style 1`] = `
.c0 {
  font-size: 2rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h3
  className="c0"
>
  Test
</h3>
`;

exports[`Typography h3 element should render as h3 variant style 1`] = `
.c0 {
  font-size: 1.75rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h3
  className="c0"
>
  Test
</h3>
`;

exports[`Typography h3 element should render as h4 variant style 1`] = `
.c0 {
  font-size: 1.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h3
  className="c0"
>
  Test
</h3>
`;

exports[`Typography h3 element should render as h5 variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h3
  className="c0"
>
  Test
</h3>
`;

exports[`Typography h3 element should render as h6 variant style 1`] = `
.c0 {
  font-size: 1rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h3
  className="c0"
>
  Test
</h3>
`;

exports[`Typography h3 element should render as lead variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  font-weight: 300;
}

<h3
  className="c0"
>
  Test
</h3>
`;

exports[`Typography h3 element should render as p variant style 1`] = `
.c0 {
  font-size: 1rem;
}

<h3
  className="c0"
>
  Test
</h3>
`;

exports[`Typography h4 element should render as h1 variant style 1`] = `
.c0 {
  font-size: 2.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h4
  className="c0"
>
  Test
</h4>
`;

exports[`Typography h4 element should render as h2 variant style 1`] = `
.c0 {
  font-size: 2rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h4
  className="c0"
>
  Test
</h4>
`;

exports[`Typography h4 element should render as h3 variant style 1`] = `
.c0 {
  font-size: 1.75rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h4
  className="c0"
>
  Test
</h4>
`;

exports[`Typography h4 element should render as h4 variant style 1`] = `
.c0 {
  font-size: 1.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h4
  className="c0"
>
  Test
</h4>
`;

exports[`Typography h4 element should render as h5 variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h4
  className="c0"
>
  Test
</h4>
`;

exports[`Typography h4 element should render as h6 variant style 1`] = `
.c0 {
  font-size: 1rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h4
  className="c0"
>
  Test
</h4>
`;

exports[`Typography h4 element should render as lead variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  font-weight: 300;
}

<h4
  className="c0"
>
  Test
</h4>
`;

exports[`Typography h4 element should render as p variant style 1`] = `
.c0 {
  font-size: 1rem;
}

<h4
  className="c0"
>
  Test
</h4>
`;

exports[`Typography h5 element should render as h1 variant style 1`] = `
.c0 {
  font-size: 2.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h5
  className="c0"
>
  Test
</h5>
`;

exports[`Typography h5 element should render as h2 variant style 1`] = `
.c0 {
  font-size: 2rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h5
  className="c0"
>
  Test
</h5>
`;

exports[`Typography h5 element should render as h3 variant style 1`] = `
.c0 {
  font-size: 1.75rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h5
  className="c0"
>
  Test
</h5>
`;

exports[`Typography h5 element should render as h4 variant style 1`] = `
.c0 {
  font-size: 1.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h5
  className="c0"
>
  Test
</h5>
`;

exports[`Typography h5 element should render as h5 variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h5
  className="c0"
>
  Test
</h5>
`;

exports[`Typography h5 element should render as h6 variant style 1`] = `
.c0 {
  font-size: 1rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h5
  className="c0"
>
  Test
</h5>
`;

exports[`Typography h5 element should render as lead variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  font-weight: 300;
}

<h5
  className="c0"
>
  Test
</h5>
`;

exports[`Typography h5 element should render as p variant style 1`] = `
.c0 {
  font-size: 1rem;
}

<h5
  className="c0"
>
  Test
</h5>
`;

exports[`Typography h6 element should render as h1 variant style 1`] = `
.c0 {
  font-size: 2.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h6
  className="c0"
>
  Test
</h6>
`;

exports[`Typography h6 element should render as h2 variant style 1`] = `
.c0 {
  font-size: 2rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h6
  className="c0"
>
  Test
</h6>
`;

exports[`Typography h6 element should render as h3 variant style 1`] = `
.c0 {
  font-size: 1.75rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h6
  className="c0"
>
  Test
</h6>
`;

exports[`Typography h6 element should render as h4 variant style 1`] = `
.c0 {
  font-size: 1.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h6
  className="c0"
>
  Test
</h6>
`;

exports[`Typography h6 element should render as h5 variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h6
  className="c0"
>
  Test
</h6>
`;

exports[`Typography h6 element should render as h6 variant style 1`] = `
.c0 {
  font-size: 1rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h6
  className="c0"
>
  Test
</h6>
`;

exports[`Typography h6 element should render as lead variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  font-weight: 300;
}

<h6
  className="c0"
>
  Test
</h6>
`;

exports[`Typography h6 element should render as p variant style 1`] = `
.c0 {
  font-size: 1rem;
}

<h6
  className="c0"
>
  Test
</h6>
`;

exports[`Typography p element should render as h1 variant style 1`] = `
.c0 {
  font-size: 2.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<p
  className="c0"
>
  Test
</p>
`;

exports[`Typography p element should render as h2 variant style 1`] = `
.c0 {
  font-size: 2rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<p
  className="c0"
>
  Test
</p>
`;

exports[`Typography p element should render as h3 variant style 1`] = `
.c0 {
  font-size: 1.75rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<p
  className="c0"
>
  Test
</p>
`;

exports[`Typography p element should render as h4 variant style 1`] = `
.c0 {
  font-size: 1.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<p
  className="c0"
>
  Test
</p>
`;

exports[`Typography p element should render as h5 variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<p
  className="c0"
>
  Test
</p>
`;

exports[`Typography p element should render as h6 variant style 1`] = `
.c0 {
  font-size: 1rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<p
  className="c0"
>
  Test
</p>
`;

exports[`Typography p element should render as lead variant style 1`] = `
.c0 {
  font-size: 1.25rem;
  font-weight: 300;
}

<p
  className="c0"
>
  Test
</p>
`;

exports[`Typography p element should render as p variant style 1`] = `
.c0 {
  font-size: 1rem;
}

<p
  className="c0"
>
  Test
</p>
`;
